src/litestorepkg/lib/api_v8.nim

```diff
@@ -0,0 +1,1407 @@
+  else:
+    result.content = $doc
+    result.code = Http200
+
+proc getStore*(LS: LiteStore, id: string, options = newQueryOptions(), req: LSRequest): LSResponse =
+  if (not LSDICT.hasKey(id)):
+    return resStoreNotFound(id)
+  let store = LSDICT[id]
+  var doc = newJObject()
+  doc["id"] = %id
+  doc["file"] = %store.file
+  doc["config"] = store.config
+  result.headers = ctJsonHeader()
+  setOrigin(LS, req, result.headers)
+  result.content = $doc
+  result.code = Http200
+
+proc getIndex*(LS: LiteStore, id: string, options = newQueryOptions(), req: LSRequest): LSResponse =
+  let doc = LS.store.retrieveIndex(id, options)
+  result.headers = ctJsonHeader()
+  setOrigin(LS, req, result.headers)
+  if doc == newJNull():
+    result = resIndexNotFound(id)
+  else:
+    result.content = $doc
+    result.code = Http200
+
+proc getRawDocument*(LS: LiteStore, id: string, options = newQueryOptions(), req: LSRequest): LSResponse =
+  let doc = LS.store.retrieveRawDocument(id, options)
+  result.headers = ctJsonHeader()
+  setOrigin(LS, req, result.headers)
+  if doc == "":
+    result = resDocumentNotFound(id)
+  else:
+    result.content = doc
+    result.code = Http200
+
+
+proc convertWikiLinks(contents: string, findDocument: proc (name: string): string, baseUrl: string): string =
+  ## replace [[Wiki Page]] links to real pages, e.g. [Wiki Page](wiki/Wiki-Page.md)   
+  let regex = re"\[\[.+?\]\]"
+  # find all wiki links and try to find corresponding documents
+  let mapping = contents.findAll(regex).deduplicate
+      .map(proc(wikiLink:string): tuple[wikiLink, link: string] =
+        let label = wikiLink[2 .. ^3]
+        let name = label.replace(" ", "-")
+        let docId = findDocument(name)
+        let link = "[" & label & "](" & baseUrl & docId & ")"
+        LOG.debug("wiki link conversion: $1 -> $2 -> $3 -> $4 -> $5", wikiLink, label, name, docId, link)
+        return (wikiLink, link)
+      )
+  
+  # replace all pairs from the mapping table
+  result = strutils.multiReplace(contents, mapping)   
+   
+
+proc renderHtml*(contents: string, getFragment: proc (name: string): string, findDocument: proc (name: string): string, baseUrl: string): string =
+  ## render markdown as HTML using HastyScribe
+  var options = HastyOptions(toc: false, output: "", css: "", watermark: "", fragment: false)
+  var fields = HastyFields()
+  var hs = newHastyScribe(options, fields)
+  # retrieve optional sidebar and footer
+  let sidebar = getFragment("sidebar").convertWikiLinks(findDocument, baseUrl)
+  let sidebarFragment = hs.compileFragment(sidebar, "")
+  let footer = getFragment("footer").convertWikiLinks(findDocument, baseUrl)
+  let footerFragment = hs.compileFragment(footer, "")
+  let linkifiedContents = contents.convertWikiLinks(findDocument, baseUrl)
+  result = hs.compileDocument(linkifiedContents, "", sidebarFragment, footerFragment)
+  
+
+proc findSpecialDocument(LS: LiteStore, dirId, name:string): string =
```

Comment:
I understand what you are trying to do here, but I am not 100% comfortable with it: you are adding a new concept of special documents for the sole purpose of adding presentation content to wiki pages. A different approach would be to store them as ordinary documents (maybe in different folders) and then have some client-side logic assemble them -- which is similar to what I am doing with the admin app, in a way.

I guess the main concept here is that LiteStore is meant to be used more as an API server than a traditional server performing server-side rendering of web pages.